src/fields/fp/u64/wrapper.rs

```diff
@@ -27,174 +25,97 @@ impl zeroize::Zeroize for Fp {
-
-        Self(x)
+        Self(ArkworksFp::from_le_bytes_mod_order(bytes))
     }
 
     pub(crate) fn to_le_limbs(&self) -> [u64; N_64] {
-        let mut x_non_montgomery = fiat::FpNonMontgomeryDomainFieldElement([0; N]);
-        fiat::fp_from_montgomery(&mut x_non_montgomery, &self.0);
-        x_non_montgomery.0
+        let le_bytes = self.to_bytes_le();
+        let mut out = [0u64; N_64];
+        for i in 0..N_64 {
+            out[i] = u64::from_le_bytes([
+                le_bytes[8 * i],
+                le_bytes[8 * i + 1],
+                le_bytes[8 * i + 2],
+                le_bytes[8 * i + 3],
+                le_bytes[8 * i + 4],
+                le_bytes[8 * i + 5],
+                le_bytes[8 * i + 6],
+                le_bytes[8 * i + 7],
+            ]);
+        }
+        out
     }
 
     pub fn to_bytes_le(&self) -> [u8; N_8] {
-        let mut bytes = [0u8; N_8];
-        let mut x_non_montgomery = fiat::FpNonMontgomeryDomainFieldElement([0; N]);
-        fiat::fp_from_montgomery(&mut x_non_montgomery, &self.0);
-        fiat::fp_to_bytes(&mut bytes, &x_non_montgomery.0);
+        let mut bytes = [0u8; 48];
+        self.0
+            .serialize_compressed(&mut bytes[..])
+            .expect("serialization into array should be infallible");
         bytes
     }
 
     pub(crate) const fn from_montgomery_limbs(limbs: [u64; N]) -> Fp {
-        Self(fiat::FpMontgomeryDomainFieldElement(limbs))
+        Self(ArkworksFp::new_unchecked(BigInt::new(limbs)))
     }
 
-    pub const ZERO: Self = Self(fiat::FpMontgomeryDomainFieldElement([0; N]));
+    pub const ZERO: Self = Self(ArkworksFp::new(BigInt::new([0; N])));
+    pub const ONE: Self = Self(ArkworksFp::new(BigInt::one()));
 
-    pub const ONE: Self = Self(fiat::FpMontgomeryDomainFieldElement([
-        202099033278250856,
-        5854854902718660529,
-        11492539364873682930,
-        8885205928937022213,
-        5545221690922665192,
-        39800542322357402,
-    ]));
-
-    pub const MINUS_ONE: Self = Self(fiat::FpMontgomeryDomainFieldElement([
+    pub const MINUS_ONE: Self = Self::from_montgomery_limbs([
         9384023879812382873,
         14252412606051516495,
         9184438906438551565,
         11444845376683159689,
         8738795276227363922,
         81297770384137296,
-    ]));
+    ]);
 
-    pub const QUADRATIC_NON_RESIDUE: Self = Self(fiat::FpMontgomeryDomainFieldElement([
+    pub const QUADRATIC_NON_RESIDUE: Self = Self::from_montgomery_limbs([
         18161750659790013178,
         10940260503947051403,
         2338003791965605956,
         14680817040264804354,
         841925479686732267,
         43193913801202386,
-    ]));
+    ]);
 
     pub fn square(&self) -> Fp {
-        let mut result = fiat::FpMontgomeryDomainFieldElement([0; N]);
-        fiat::fp_square(&mut result, &self.0);
-        Self(result)
+        Fp(self.0.square())
     }
 
     pub fn inverse(&self) -> Option<Self> {
```

Comment:
the `fiat-crypto` generated `inverse` implementation was extremely slow (~8x) compared to arkworks in 64-bit backend.